### ISSUE TITLE
Actions: Support continue-on-error

### ISSUE THREAD

**Issue by bilderbuchi** [go-gitea/gitea#30389]
### Feature Description

As per the docs
https://github.com/go-gitea/gitea/blob/50099d7af436785daf66a3a9f27bd5c009f90684/docs/content/usage/actions/comparison.en-us.md?plain=1#L64-L68
, `continue-on-error` is currently ignored , but I could not find an issue tracking its addition/implementation.

Context:
I would like to make problems found by linters or formatters visible in the build status/output, without manually opening and reading step logs and without terminating other build steps (e.g. subsequent testsuite runs) if such problems are present.

As problem matchers are _also_ not supported yet (#28452), annotations are also not supported yet (#237229) and a "warning"/yellow build status, like e.g. buildbot has, is also not supported, `continue-on-error` is the only way I could come up with to flag linting/formatting issues without terminating the build outright.

Thank you for your consideration!

**Comment by imkuang** [comment]
Hi,

I had been using `continue-on-error` in my workflow before I noticed this document. It worked well for me until today when I discovered that the document say Gitea Action currently does not support this syntax :( ...
Something must have gone wrong. I wonder if we have already supported this feature, but the document has not been updated?
<img width="355" alt="image" src="https://github.com/go-gitea/gitea/assets/38498747/ee64d06f-0ef1-4606-ae06-fa7a80ad59bc">
My Gitea version is 1.21.10

**Comment by imkuang** [comment]
Yes, the overall result of that job is passed, and you need to manually check the build step log to find out that this step has failed.
I think this should be the normal behavior of `continue-on-error`, but it indeed seems not to achieve your purpose.

**Comment by bcbnz** [comment]
I have found another potential workaround: add `if: always()` to the steps after the one that might fail. They will always run, and the result of the job will be marked as failed if any of them fail. Works for me on Gitea 1.22.1.

**Comment by merlinz01** [comment]
Another workaround would be to put ` || true` after any command you want to be able to fail without stopping the build. This would not show the red X icon for the step though.

**Comment by VAllens** [comment]
It took me this long to notice that `continue-on-error` wasn't working in the workflow based on Gitea Actions.
Are there any plans to support it?

**Comment by bircni** [comment]
this was implemented and added with gitea 1.27
